server/api/core/core.js

```diff
@@ -311,60 +303,52 @@ export default {
         return profile.preferences[packageName][preference];
       }
     }
-
     return defaultValue || undefined;
   },
 
-  /** // TODO: Check to see if this is still being used
-   * Add default roles for new visitors
-   * @param {String|Array} roles - A string or array of roles and routes
-   * @returns {undefined} - does not specifically return anything
-   */
-  addDefaultRolesToVisitors(roles) {
-    Logger.info(`Adding defaultRoles & defaultVisitorRole permissions for ${roles}`);
-
-    const shop = Shops.findOne(this.getShopId());
-
-    if (Match.test(roles, [String])) {
-      Shops.update(shop._id, {
-        $addToSet: { defaultVisitorRole: { $each: roles } }
-      });
-      Shops.update(shop._id, {
-        $addToSet: { defaultRoles: { $each: roles } }
-      });
-    } else if (Match.test(roles, String)) {
-      Shops.update(shop._id, {
-        $addToSet: { defaultVisitorRole: roles }
-      });
-      Shops.update(shop._id, {
-        $addToSet: { defaultRoles: roles }
-      });
-    } else {
-      throw new Meteor.Error(`Failed to add default roles ${roles}`);
-    }
-  },
-
   /**
-   * Add default roles for new sellers
-   * @param {String|Array} roles A string or array of roles and routes
-   * @returns {undefined} - does not specifically return anything
+   *  insertPackagesForShop
+   *  insert Reaction packages into Packages collection registry for a new shop
+   *  Assigns owner roles for new packages
+   *  Imports layouts from packages
+   *  @param {String} shopId - the shopId you need to create packages for
+   *  @return {String} returns insert result
    */
-  addDefaultRolesToSellers(roles) {
-    Logger.info(`Adding defaultSellerRoles permissions for ${roles}`);
+  insertPackagesForShop(shopId) {
+    const layouts = [];
+    if (!shopId) {
+      return [];
+    }
+    const packages = this.Packages;
+    // for each shop, we're loading packages in a unique registry
+    // Object.keys(pkgConfigs).forEach((pkgName) => {
+    for (const packageName in packages) {
```

Comment:
actually, packages is an object where the keys are the packageName, not an array.